source/code/include/playfab/PlayFabPlatformUtils.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <sstream>
+
+namespace PlayFab
+{
+    typedef std::chrono::time_point<std::chrono::system_clock> TimePoint;
+
+    inline TimePoint GetTimePointNow()
+    {
+        return std::chrono::system_clock::now();
+    }
+
+    inline time_t GetPlayFabTimeTNow()
+    {
+        time_t now = std::chrono::system_clock::to_time_t(GetTimePointNow());
+#if defined(PLAYFAB_PLATFORM_IOS) || defined(PLAYFAB_PLATFORM_ANDROID) || defined(PLAYFAB_PLATFORM_LINUX) || defined(PLAYFAB_PLATFORM_SWITCH)
+        return static_cast<Json::Int64>(now);
+#else // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_SWITCH
+        return now;
+#endif // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_SWITCH
+    }
+
+    inline void AppendIntToString(int value, std::string& output)
+    {
+        // itoa is not available in android
+        char buffer[16];
+#if defined(PLAYFAB_PLATFORM_IOS) || defined(PLAYFAB_PLATFORM_ANDROID) || defined(PLAYFAB_PLATFORM_LINUX) || defined(PLAYFAB_PLATFORM_PLAYSTATION) || defined(PLAYFAB_PLATFORM_SWITCH)
+        sprintf(buffer, "%d", value);
+#else // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_PLAYSTATION || PLAYFAB_PLATFORM_SWITCH
+        sprintf_s(buffer, "%d", value);
+#endif // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_PLAYSTATION || PLAYFAB_PLATFORM_SWITCH
+        output.append(buffer);
+    }
+
+    inline std::string LocalTimeTToUtcString(time_t now)
+    {
+        tm timeInfo;
+#if defined(PLAYFAB_PLATFORM_WINDOWS) || defined(PLAYFAB_PLATFORM_XBOX)
+        gmtime_s(&timeInfo, &now);
```

Comment:
>gmtime_s [](start = 8, length = 8)

the last time I brought up asserting this possible error return you responded by saying that it'd be a "whole new feature" to assert this return's success...

I don't think I understand. the c and c++ std have utilities for performing asserts. I would not considering adding a single line a new feature

```
errno_t err = gmtime_s(...);
std::assert(err == 0);
```

but maybe I misunderstood. what did you mean?

Reply:
Yes, but we haven't discussed the ramifications of throwing asserts in code that customers call.

Reply:
you already have asserts in customer code. check out PlayFabBaseModel.h line 25
`inline operator const BoxedType& () const { assert(notNull()); return *operator->(); }`